### PR TITLE
MOSIP-32460 Merge java 21 branch to develop

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

File: id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/CredentialRequestGeneratorBootApplication.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import io.mosip.kernel.dataaccess.hibernate.repository.impl.HibernateRepositoryImpl;
+
+/**
+ * The Class CredentialRequestGeneratorApp.
+ *
+ */
+@SpringBootApplication
+@Import(value = { java.lang.String.class, DummyPartnerCheckUtil.class, RestHelper.class, IdRepoSecurityManager.class,
+		CredentialRequestGeneratorConfig.class })
+@ComponentScan(basePackages = { "io.mosip.*", "${mosip.auth.adapter.impl.basepackage}" }, excludeFilters = {
```
Comment (dhanendra06) [inline id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/CredentialRequestGeneratorBootApplication.java:28]: please scan specific libraries that required

File: id-repository/id-repository-credentials-feeder/pom.xml [hunk 2]
```diff
@@ -10,7 +10,7 @@
 	<parent>
 		<groupId>io.mosip.idrepository</groupId>
 		<artifactId>id-repository-parent</artifactId>
-		<version>1.2.1-SNAPSHOT</version>
+		<version>1.2.0.1</version>
```
Comment (dhanendra06) [inline id-repository/id-repository-credentials-feeder/pom.xml:13]: why idrepo-parent version is 1.2.0.1

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/IdRepoBootApplication.java [hunk 3]
```diff
@@ -5,13 +5,18 @@
...
+import io.mosip.idrepository.identity.config.IdRepoConfig;
 
 /**
  * The Class IdRepoApplication.
  *
  * @author Manoj SP
  */
 @SpringBootApplication(exclude = MailSenderAutoConfiguration.class)
+@Import({ IdRepoConfig.class })
 @ComponentScan(basePackages={ "io.mosip.*" ,"${mosip.auth.adapter.impl.basepackage}"}, excludeFilters = {
```
Comment (dhanendra06) [inline id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/IdRepoBootApplication.java:20]: please scan specific libraries

File: id-repository/credential-request-generator/pom.xml [hunk 4]
```diff
@@ -249,29 +188,28 @@
...
 			<artifactId>reactor-core</artifactId>
-			<version>3.1.6.RELEASE</version>
 		</dependency>
 		<dependency>
 			<groupId>org.springframework</groupId>
 			<artifactId>spring-jdbc</artifactId>
-			<version>4.3.8.RELEASE</version>
 		</dependency>
 		<dependency>
 			<groupId>org.springdoc</groupId>
```
Comment (dhanendra06) [inline id-repository/credential-request-generator/pom.xml:201]: springdoc-openapi-ui is not required if you add springdoc-openapi-starter-webmvc-ui

**Review by dhanendra06 — CHANGES_REQUESTED** [review]

File: id-repository/id-repository-credentials-feeder/pom.xml (hunk 2 above)
Comment (kameshsr) [reply id-repository/id-repository-credentials-feeder/pom.xml:13]: Credential feeder is not migrated so using idrepo parent 1201

File: id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/CredentialRequestGeneratorBootApplication.java (hunk 1 above)
Comment (kameshsr) [reply id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/CredentialRequestGeneratorBootApplication.java]: DOne.

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/IdRepoBootApplication.java (hunk 3 above)
Comment (kameshsr) [reply id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/IdRepoBootApplication.java:20]: Done.

File: id-repository/credential-request-generator/pom.xml (hunk 4 above)
Comment (kameshsr) [reply id-repository/credential-request-generator/pom.xml]: Corrected.

File: id-repository/id-repository-credentials-feeder/pom.xml (hunk 2 above)
Comment (kameshsr) [reply id-repository/id-repository-credentials-feeder/pom.xml:13]: [MOSIP-35146](https://mosip.atlassian.net/browse/MOSIP-35146) 
https://mosip.atlassian.net/browse/MOSIP-35147
jira tickets attatched.

**Review by dhanendra06 — APPROVED** [review]

File: id-repository/id-repository-credentials-feeder/src/test/java/io/mosip/idrepository/credentialsfeeder/step/test/CredentialsFeedingWriterTest.java [hunk 5]
```diff
@@ -48,6 +49,7 @@
 @WebMvcTest
 @Import(EnvUtil.class)
 @ActiveProfiles("test")
+@Ignore
```
Comment (vishwa-vyom) [inline id-repository/id-repository-credentials-feeder/src/test/java/io/mosip/idrepository/credentialsfeeder/step/test/CredentialsFeedingWriterTest.java:52]: why are we ignoring the full test case files?

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: id-repository/id-repository-credentials-feeder/src/test/java/io/mosip/idrepository/credentialsfeeder/step/test/CredentialsFeedingWriterTest.java (hunk 5 above)
Comment (kameshsr) [reply id-repository/id-repository-credentials-feeder/src/test/java/io/mosip/idrepository/credentialsfeeder/step/test/CredentialsFeedingWriterTest.java]: Fixed it.

File: db_upgrade_scripts/mosip_credential/sql/1.2.0.1_to_1.2.1.0-B1_rollback.sql [hunk 6]
```diff
@@ -0,0 +1,22 @@
+-- ------------------------------------------------------------------------------------------
+-- Revoke script for Migrating Spring batch version to 5.0 as part of Java 21 Migration.
+-- References: 
+--  1. https://github.com/spring-projects/spring-batch/wiki/Spring-Batch-5.0-Migration-Guide#ms-sqlserver
+--  2. https://github.com/spring-projects/spring-batch/blob/main/spring-batch-core/src/main/resources/org/springframework/batch/core/migration/5.0/migration-postgresql.sql
+-- ------------------------------------------------------------------------------------------
+ALTER TABLE BATCH_STEP_EXECUTION DROP CREATE_TIME TIMESTAMP NOT NULL DEFAULT '1970-01-01 00:00:00';
```
[the remark this answers is not in the document]
Comment (kameshsr) [reply db_upgrade_scripts/mosip_credential/sql/1.2.0.1_to_1.2.1.0-B1_rollback.sql]: Changed file name to 1.2.0.1_to_1.3.0_ugrade.sql and rollback

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**